Tallies/TallyClerks/MGxsClerk_class.f90

```diff
@@ -0,0 +1,887 @@
+
+    ! Get current particle state
+    state = p
+
+    ! Find bin indexes
+    ! Energy
+    if (allocated(self % energyMap)) then
+      enIdx = self % energyN + 1 - self % energyMap % map(state)
+    else
+      enIdx = 1
+    end if
+    ! Space
+    if (allocated(self % spaceMap)) then
+      matIdx = self % spaceMap % map(state)
+    else
+      matIdx = 1
+    end if
+
+    ! Return if invalid bin index
+    if (enIdx == 0 .or. matIdx == 0) return
+
+    ! Calculate bin address
+    binIdx = self % energyN * (matIdx - 1) + enIdx
+    addr = self % getMemAddress() + self % width * (binIdx - 1) - 1
+
+    ! Get material pointer
+    mat => neutronMaterial_CptrCast(xsData % getMaterial(p % matIdx()))
+    if (.not.associated(mat)) then
+      call fatalError(Here,'Unrecognised type of material was retrived from nuclearDatabase')
+    end if
+
+    ! Retrieve material cross sections
+    call mat % getMacroXSs(xss, p)
+
+    ! Calculate flux
+    totalXS  = xss % total
+    flux = p % w / totalXS
+
+    ! Calculate reaction rates
+    nuFissXS = xss % nuFission * flux
+    captXS   = xss % capture * flux
+    fissXS   = xss % fission * flux
+    scattXS  = (xss % elasticScatter + xss % inelasticScatter) * flux
+
+    ! Add scores to counters
+    call mem % score(flux,     addr + FLUX_idx)
+    call mem % score(nuFissXS, addr + NUBAR_idx)
+    call mem % score(captXS,   addr + CAPT_idx)
+    call mem % score(fissXS,   addr + FISS_idx)
+    call mem % score(scattXS,  addr + SCATT_idx)
+
+  end subroutine reportInColl
+
+  !!
+  !! Process outgoing collision report
+  !!
+  !! See tallyClerk_inter for details
+  !!
+  subroutine reportOutColl(self, p, MT, muL, xsData, mem)
+    class(MGxsClerk), intent(inout) :: self
+    class(particle), intent(in)             :: p
+    integer(shortInt), intent(in)           :: MT
+    real(defReal), intent(in)               :: muL
+    class(nuclearDatabase),intent(inout)    :: xsData
+    type(scoreMemory), intent(inout)        :: mem
+    type(particleState)                     :: preColl, postColl
+    real(defReal)                           :: score, prod, mu, mu2, mu3, mu4, mu5
+    integer(shortInt)                       :: enIdx, matIdx, binIdx, binEnOut
+    integer(longInt)                        :: addr
+
+    ! Get pre and post collision particle state
+    preColl  = p % preCollision
+    postColl = p
+
+    ! Find multipliers to account for scattering multiplicity
+    select case(MT)
+      case(N_2N, N_2Na, N_2Nd, N_2Nf, N_2Np, N_2N2a, N_2Nl(1):N_2Nl(16))
+        score = 2.0_defReal
+      case(N_3N, N_3Na, N_3Nf, N_3Np)
+        score = 3.0_defReal
+      case(N_4N)
+        score = 4.0_defReal
+      case default
+        score = ONE
+    end select
+
+    ! Score in case of scattering events
+    select case(MT)
+      case ( N_N_ELASTIC, N_N_INELASTIC, N_Nl(1):N_Nl(40), N_Ncont,         &
+             N_2N, N_2Na, N_2Nd, N_2Nf, N_2Np, N_2N2a, N_2Nl(1):N_2Nl(16),  &
```

Comment:
Out of my personal curiosity: 
Do `N_2N1` - `N_2N16` ever get sampled in transport calculations? 
I always though they were covered with MT=16